src/tardev-snapshotter/tarindex/src/lib.rs

```diff
@@ -136,18 +137,13 @@ fn read_all_entries(
                             .link_name_bytes()
                             .unwrap_or(std::borrow::Cow::Borrowed(b""));
                         if *hname != *name {
-                            // TODO: Handle this case by duplicating the full name.
-                            eprintln!(
-                                "Skipping symlink with long link name ({}, {} bytes, {}, {} bytes): {}",
-                                String::from_utf8_lossy(&name), name.len(),
-                                String::from_utf8_lossy(&hname), hname.len(),
-                                String::from_utf8_lossy(&f.path_bytes())
-                            );
-                            continue;
+                            special_link.push(String::from_utf8_lossy(&name).to_string());
+                            entry_offset = 0;
+                        }
+                        else {
```

Comment:
Did you run cargo fmt? I think it would merge the two lines above into a single.

Reply:
Good catch! I forgot to run cargo fmt. Just pushed its format changes.